searchsummary/src/vespa/searchsummary/docsummary/keywordextractor.cpp

```diff
@@ -42,32 +41,25 @@ KeywordExtractor::IsLegalIndexName(const char *idxName) const
 }
 
 KeywordExtractor::IndexPrefix::IndexPrefix(const char *prefix, IndexPrefix **list)
-    : _prefix(NULL),
-      _prefixLen(0),
-      _next(NULL)
+    : _prefix(prefix),
+      _next(nullptr)
 {
-    _prefix = strdup(prefix);
-    assert(_prefix != NULL);
-    _prefixLen = strlen(prefix);
     _next = *list;
     *list = this;
 }
 
-KeywordExtractor::IndexPrefix::~IndexPrefix()
-{
-    free(_prefix);
-}
+KeywordExtractor::IndexPrefix::~IndexPrefix() = default;
 
 bool
 KeywordExtractor::IndexPrefix::Match(const char *idxName) const
 {
-    return (strncmp(idxName, _prefix, _prefixLen) == 0);
+    return (strncmp(idxName, _prefix.c_str(), _prefix.size()) == 0);
```

Comment:
should use a string compare function here

Reply:
Fixed.